pyintegration/deephaven2/stream/kafka/producer.py

```diff
@@ -0,0 +1,195 @@
+            Kafka messages.  This should be the result of calling one of the methods
+            simple, avro or json in this module, or the constant IGNORE.
+        last_by_key_columns:  Whether to publish only the last record for each unique key.
+            Ignored if key is IGNORE.  If key is not IGNORE and last_by_key_columns is false,
+            it is expected that table updates will not produce any row shifts; that is, the publisher
+            expects keyed tables to be streams, add-only, or aggregated.
+
+    Returns:
+        a callback that, when invoked, stops publishing and cleans up subscriptions and resources.
+        Users should hold to this callback to ensure liveness for publishing for as long as this
+        publishing is desired, and once not desired anymore they should invoke it.
+
+    Raises:
+        ValueError, TypeError, DHError
+    """
+
+    if not isinstance(topic, str):
+        raise ValueError("argument 'topic' has to be of str type, instead got " + topic)
+
+    if key is None:
+        raise ValueError("argument 'key' is None")
+    if value is None:
+        raise ValueError("argument 'value' is None")
+    if key is IGNORE and value is IGNORE:
+        raise ValueError(
+            "at least one argument for 'key' or 'value' must be different from IGNORE")
+
+    kafka_config = _dict_to_j_properties(kafka_config)
+    try:
+        runnable = _JKafkaTools.produceFromTable(table.j_table, kafka_config, topic, key, value, last_by_key_columns)
+    except Exception as e:
+        raise DHError(e, "failed to start producing Kafka messages.") from e
+
+    def cleanup():
+        runnable.run()
+
+    return cleanup
+
+
+def avro(schema, schema_version: str = None, field_to_col_mapping=None, timestamp_field=None):
+    """ Specify an Avro schema to use when producing a Kafka stream from a Deephaven table.
+
+    Args:
+        schema:  either an Avro schema object or a string specifying a schema name for a schema
+            registered in a Confluent compatible Schema Server.  When the latter is provided, the
+            associated kafka_config dict in the call to consumeToTable should include the key
+            'schema.registry.url' with the associated value of the Schema Server URL for fetching the schema
+            definition.
+        schema_version:  if a string schema name is provided, the version to fetch from schema
+            service; if not specified, a default of 'latest' is assumed.
+        field_to_col_mapping: a dict mapping field names in the schema to column names in the Deephaven table.
+            Any fields in the schema not present in the dict as keys are mapped to columns of the same name.
+            If this argument is None, all schema fields are mapped to columns of the same name.
+        timestamp_field: a string for the name of an additional timestamp field to include,
+            or None for no such field.
```

Comment:
Generally, we always want to have some kind of timestamp for the stages we are processing.  Having timestamps in the payload instead of just things in a header is also sometimes beneficial when headers get stripped away, because your Avro data may be destined for places other than Kafka.

Reply:
Sounds like we just need very clear pydocs.